kuma/wiki/views/edit.py

```diff
@@ -210,36 +209,54 @@ def edit(request, document_slug, document_locale, revision_id=None):
+                        rev_form.errors['current_rev'][0] = mark_safe(rev_form.errors['current_rev'][0])
+
+                        # If this was an Ajax POST, then return a JsonResponse
+                        if is_async_submit:
+                            data = {
+                                "error": True,
+                                "error_message": rev_form.errors['current_rev'],
+                                "new_revision_id": curr_rev.id,
+                            }
+                            return JsonResponse(data=data, status=409)
                         # Jump out to a function to escape indentation hell
                         return _edit_document_collision(
-                            request, orig_rev, curr_rev, is_iframe_target,
+                            request, orig_rev, curr_rev, is_async_submit,
                             is_raw, rev_form, doc_form, section_id,
                             rev, doc)
-
+                    # Was this an Ajax submission that was marked as spam?
+                    if is_async_submit and '__all__' in rev_form._errors:
+                        # Return a JsonResponse
+                        data = {
+                            "error": True,
+                            "error_message": mark_safe(rev_form.errors['__all__'][0]),
+                            "new_revision_id": curr_rev.id,
+                        }
+                        return JsonResponse(data=data)
                 if rev_form.is_valid():
                     rev_form.save(doc)
-
-                    if is_iframe_target:
-                        # TODO: Does this really need to be a template? Just
-                        # shoehorning data into a single HTML element.
-                        response = HttpResponse("""
-                            <span id="iframe-response"
-                                  data-status="OK"
-                                  data-current-revision="%s">OK</span>
-                        """ % doc.current_revision.id)
-                        response['X-Frame-Options'] = 'SAMEORIGIN'
-                        return response
-
-                    if (is_raw and orig_rev is not None and
-                            curr_rev.id != orig_rev.id):
-                        # If this is the raw view, and there was an original
-                        # revision, but the original revision differed from the
-                        # current revision at start of editing, we should tell
-                        # the client to refresh the page.
-                        response = HttpResponse('RESET')
-                        response['X-Frame-Options'] = 'SAMEORIGIN'
-                        response.status_code = 205
-                        return response
+                    # Is this an Ajax POST?
+                    if is_async_submit:
+                        if (is_raw and orig_rev is not None and
```

Comment:
Before, this code ran both for normal "save" and for the AJAX "save and keep editing", and now it just runs for the AJAX version. Was this intentional?